src/oci.c

```diff
@@ -216,6 +216,61 @@ cc_oci_get_config_and_state (gchar **config_file,
 	return false;
 }
 
+
+gboolean
+cc_oci_create_cgroups (struct cc_oci_config *config)
+{
+	gchar *cgroup_dir = NULL;
+	gchar *cgroup_tasks = NULL;
+	gchar *cgroup_procs = NULL;
+	char  *pid_str = NULL;
+	int   fd_tasks = -1;
+	int   fd_procs = -1;
+
+	if (!config) {
+		return false;
+	}
+
+	if (!config->oci.oci_linux.cgroupsPath) {
+		return true;
+	}
+
+	cgroup_dir = g_strdup_printf ("%s/%s", CGROUP_MEM_DIR,
+				      config->oci.oci_linux.cgroupsPath);
+	g_mkdir_with_parents (cgroup_dir, CC_OCI_CGROUP_MODE);
+
+	pid_str = g_strdup_printf ("%d", config->state.workload_pid);
+
+	cgroup_tasks = g_strdup_printf ("%s/tasks", cgroup_dir);
+	fd_tasks = open (cgroup_tasks, O_WRONLY);
+	if (write (fd_tasks, pid_str, strlen(pid_str)) <= 0) {
+		g_critical ("failed to copy workload pid to cgroup tasks: %s",
+			    strerror(errno));
+		goto out;
+	}
+
+	cgroup_procs = g_strdup_printf ("%s/cgroup.procs", cgroup_dir);
+	fd_procs = open (cgroup_procs, O_WRONLY);
+	if (write (fd_procs, pid_str, strlen(pid_str)) <= 0) {
+		g_critical ("failed to copy workload pid to cgroup procs: %s",
+			    strerror(errno));
+		goto out;
+	}
+
+	return true;
```

Comment:
you can't ```return true``` , you have to free memory and close open fds

you should add a new variable ```bool ret=false;``` and replace this line with ```ret=true;```